scargo/scargo_src/sc_docker.py

```diff
@@ -41,61 +42,68 @@ def scargo_docker(
     docker_path = Path(project_path, ".devcontainer")
 
     if build_docker:
-        _scargo_build_docker(docker_path, no_cache)
+        _scargo_build_docker(docker_path, docker_opts=docker_opts)
 
     if run_docker:
-        _scargo_run_docker(docker_path, project_config)
+        _scargo_run_docker(docker_path, project_config, docker_opts=docker_opts)
 
     if exec_docker:
-        _scargo_exec_docker(project_config)
+        _scargo_exec_docker(project_config, docker_opts=docker_opts)
 
 
-def _scargo_build_docker(docker_path: Path, no_cache: bool = False) -> None:
+def _scargo_build_docker(docker_path: Path, docker_opts: Sequence = tuple()) -> None:
     """
     Build docker
 
     :param Path docker_path: path to dockerfile
-    :param bool no_cache: with/without docker cache
+    :param docker_opts: additional docker options
     :raises CalledProcessError: if docker build fail
     """
     logger = get_logger()
     logger.debug("Build docker environment.")
 
-    cache = "--no-cache" if no_cache else ""
+    cmd = " ".join(["docker-compose build", *docker_opts])
+
     try:
-        subprocess.run(f"docker-compose build {cache}", shell=True, cwd=docker_path)
+        subprocess.run(cmd, shell=True, cwd=docker_path, check=False)
```

Comment:
We should add check=True to all subprocess.run calls:
`If check is true, and the process exits with a non-zero exit code, a [CalledProcessError](https://docs.python.org/3/library/subprocess.html#subprocess.CalledProcessError) exception will be raised. Attributes of that exception hold the arguments, the exit code, and stdout and stderr if they were captured.`